### PR TITLE
Add Dequantize OP

### PR DESCRIPTION
**Pull request by xiaolil1**
Split from #14929 
test=develop

Quantize OP is in #14974

### CODE REVIEW

File: paddle/fluid/operators/dequantize_mkldnn_op.cc [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    auto* input = ctx.Input<Tensor>("Input");
+    auto scale_data = ctx.Attr<float>("Scale");
+    auto* output = ctx.Output<Tensor>("Output");
+    auto& dev_ctx =
+        ctx.template device_context<platform::MKLDNNDeviceContext>();
+    const auto& engine = dev_ctx.GetEngine();
+
+    const T* input_data = input->data<T>();
+    float* output_data = output->mutable_data<float>(ctx.GetPlace());
+    std::vector<float> reorder_scale = {1.0f / scale_data};
```
Comment (tpatejko) [inline paddle/fluid/operators/dequantize_mkldnn_op.cc:46]: What could the `T` type be? Could it be either `int8_t` or `uint8_t`? Can `output_data` be also `double` or just `float`?

Comment (xiaolil1) [reply paddle/fluid/operators/dequantize_mkldnn_op.cc:46]: T could be int8_t and uint8_t, and output should only be float.

Comment (tpatejko) [reply paddle/fluid/operators/dequantize_mkldnn_op.cc:46]: @xiaolil1 Ok, understood!

**Comment by tpatejko** [comment]
@xiaolil1 @luotao1 LGTM!

**Review by tpatejko — DISMISSED** [review]

**Review by Sand3r- — DISMISSED** [review]
LGTM

**Review by tensor-tang — APPROVED** [review]
LGTM

**Pull request merged**